### PR TITLE
Attn analysis

### PR DESCRIPTION
**Pull request by Zahra-Alavi**
I added a bunch of scripts for analyzing attention, but I mainly need the regression implementation reviewed. Thanks!

### CODE REVIEW

File: scripts/final_pipeline/models.py [hunk 1]
```diff
@@ -255,5 +235,177 @@ def forward(self, inputs, targets):
...
+        emb = self.embedding_model(input_ids=input_ids, attention_mask=attention_mask).last_hidden_state
+        h, _ = self.lstm(emb)
+        h = self.dropout(h)
+        output = self.fc(h)  # [B, L, num_outputs]
+        
+        feats = h if return_features == "pre" else (output if return_features == "post" else None)
+        return output, feats
+
+
+class BiLSTMWithSelfAttentionRegressionModel(nn.Module):
```
Comment (mkorpusik) [inline scripts/final_pipeline/models.py:283]: A lot of this model code looks like it's the same as the classification models' code, just with num_outputs instead of num_classes in the final linear layer. Maybe we could combine to simplify/shorten the code? But it looks right!

**Review by mkorpusik — APPROVED** [review]
The regression models and training/evaluation look good to me! It sounded like you were getting pretty good MSE scores, so I guess the question is how close to the true Neq values the predicted regression values should be for us to use our model's predictions instead of needing to run MD simulations? Also, it's not typical to compare classification to regression, but it could be an interesting sanity check for us to use the classification thresholds to discretize the predicted regression values and then compare to the classification models on test set precision, recall, and F1.

**Comment by Zahra-Alavi** [comment]
There was an issue with regression implementation, padded amino acids were not being handled properly and were included in evaluation which is why the performance was so good! I fixed that and the performance is awful now! :D
